dlt/pipeline/pipeline.py

```diff
@@ -170,6 +171,24 @@ def _wrap(self: "Pipeline", *args: Any, **kwargs: Any) -> Any:
     return _wrap  # type: ignore
 
 
+def with_plugins_step() -> Callable[[TFun], TFun]:
+    def decorator(f: TFun) -> TFun:
+        @wraps(f)
+        def _wrap(self: "Pipeline", *args: Any, **kwargs: Any) -> Any:
+            plugins_context = self._container[PluginsContext]
```

Comment:
this must be implement the same way `with_state` is implemented. you need to inject `PluginContext` in the container (so you create instance here, pass a list of plugins, do the setup, I hope it is not too heavy.)

btw. I still do not understand if `Plugin` instance is supposed to survive steps? so for example if someone is sending contract violations to a kafka topic, the Kafka client should be created for each step?

Reply:
I built in a way now that is survives the steps. I also don't really know which approach is right here, but we can do it like this for now and see where we get when implementing new stuff.